### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by mrhoribu** [elanthia-online/illthorn#191]
Accessibility screen reader support. GM Avaluka was discussing accessibility FE clients and briefly mentioned to her about Illthorn being Electron based. She mentioned it should be fairly easy to add screen reader support via WAI-ARIA markup if we had an option to include it. Not sure if maybe creating a theme/layout for accessibility would be best. Right now visual impaired players have very few options to choose from.

https://webaim.org/techniques/aria/

Avaluka mentioned placing `role="log"` in the div tag on the story window equivalent would likely fix the #1 issue with any FE for a screen reader ... new text being read out automatically.

**Comment by ondreian** [comment]
This is definitely a design goal, I'm glad she mentioned/shared which role was best because I honestly had no idea and that was one of the biggest hurdles.

**Comment by vaneseltine** [comment]
A couple other things I noted from the conversation with Avaluka:

1. She uses Jaws (which is not free) and mentioned NVDA as a decent free alternative for testing purposes, though both are Windows-only. https://www.nvaccess.org/download/
2. She described the login process as a "main hurdle" of working with Lich, as its GUI is unfriendly and command line stuff is generally difficult. I see `:launch` is already in progress (#186)

It sounds like Illthorn has a lot of potential as an accessible FE.

**Comment by ondreian** [comment]
I do not have a windows machine, but there is a fairly comprehensive test suite for a11y systems that I often use for $WORK$, not quite sure how I can plug it into Electron yet, but I guess it will be fun to figure out.

I would like to provide wrappers around most scripting engine stuff inside of Illthorn, but I have to finish up some crucial things on Urnon first.

**Comment by vaneseltine** [comment]
Cool!

**Comment by chriscoyier** [comment]
Just noting that having a REAL PERSON to test this is fantastic. 

> Not sure if maybe creating a theme/layout for accessibility would be best.

Every theme should be accessible. The differences in themes are largely very cosmetic even now and should have very little if any impact on screen-reader accessibility. Accessibility involves stuff like color contrast too, so that's another ball game, but pure screen-reader stuff should be pretty identical.

---

This would be awfully need to jack into Electron in dev-mode if possible: 
https://www.deque.com/axe/devtools/

**Comment by mrhoribu** [comment]
Without getting feedback from an actual blind person, not sure if having a minimal theme would be good or not. Just not sure how much "stuff" should be on the screen for the screen reader to get overwhelmed with to read was my idea of doing a specific theme/layout. I do agree they should all be accessible though.

I know Avaluka did say one of the hardest parts right now is basic character creation and logging in. She mentioned the current Lich GUI launcher is awful for a screen reader to use. So much that she had to get her visual friends to help her create new logins in it to use. I'm not sure if this is partly due to current Lich GUI launcher being GTK2 and if GTK3 supports additional accessibility options, but that's likely something to push over on the GTK3 Lich or Urnon dev repository.
